include/deal.II/lac/trilinos_tpetra_vector.templates.h

```diff
@@ -289,7 +289,43 @@ namespace LinearAlgebra
       //  - Third case: the vectors have different size.
       if (vector->getMap()->isSameAs(*V.vector->getMap()))
         {
-          *vector = *V.vector;
+          // Create a read-only Kokkos view from the source vector
+#  if DEAL_II_TRILINOS_VERSION_GTE(13, 2, 0)
+          auto source_vector_2d =
+            V.vector->template getLocalView<Kokkos::HostSpace>(
+              Tpetra::Access::ReadOnly);
+#  else
+          auto source_vector_2d =
+            V.vector->template getLocalView<Kokkos::HostSpace>();
+#  endif
```

Comment:
Since we are now using `deep_copy`, is there a good reason to extract the host view instead of the device view? In case of a device memory space (which we don't support yet), the approach here does unnecessary copies between host and device since all operations are explicitly done on the host.

Reply:
Let's just get this in and we can look at this somewhere else.